gobblin-metrics/src/test/java/gobblin/metrics/performance/PerformanceUtils.java

```diff
@@ -0,0 +1,204 @@
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+
+import com.google.common.base.Strings;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import com.google.common.collect.Sets;
+
+
+/**
+ * Methods to run performance tests on Gobblin Metrics.
+ */
+@Slf4j
+public class PerformanceUtils {
+
+  /**
+   * Runs a set of performance tests. The method will take the cardinal product of the values of each input parameter,
+   * and run a performance test for each combination of paramters. At the end, it will print out the results.
+   *
+   * <p>
+   *   All parameters (except for queries) are a set of integers, meaning that separate tests will be run for all
+   *   the values provided. The number of queries will be identical for all tests.
+   * </p>
+   *
+   * @param threads Number of threads to spawn. Each thread will have an {@link Incrementer} and update metrics.
+   * @param depth Depth of the {@link gobblin.metrics.MetricContext} tree.
+   * @param forkAtDepth If multiple threads, each thread has its own {@link gobblin.metrics.MetricContext}. This
+   *                    parameter sets the first level in the tree where the per-thread MetricContexts branch off.
+   * @param counters Number of counters to generate per thread.
+   * @param meters Number of meters to generate per thread.
+   * @param histograms Number of histograms to generate per thread.
+   * @param timers Number of timers to generate per thread.
+   * @param queries Number of increments to do, divided among all threads.
+   * @throws Exception
+   */
+  @Builder(buildMethodName = "run", builderMethodName = "multiTest")
+  public static void _multiTest(@Singular("threads") Set<Integer> threads,
+      @Singular("depth") Set<Integer> depth,
+      @Singular("forkAtDepth") Set<Integer> forkAtDepth,
+      @Singular("counters") Set<Integer> counters,
+      @Singular("meters") Set<Integer> meters,
+      @Singular("histograms") Set<Integer> histograms,
+      @Singular("timers") Set<Integer> timers,
+      long queries, String name) throws Exception {
+
+    if(threads.isEmpty()) {
+      threads = Sets.newHashSet(1);
+    }
+    if(forkAtDepth.isEmpty()) {
+      forkAtDepth = Sets.newHashSet(0);
+    }
+    if(depth.isEmpty()) {
+      depth = Sets.newHashSet(0);
+    }
+    if(counters.isEmpty()) {
+      counters = Sets.newHashSet(0);
+    }
+    if(meters.isEmpty()) {
+      meters = Sets.newHashSet(0);
+    }
+    if(histograms.isEmpty()) {
+      histograms = Sets.newHashSet(0);
+    }
+    if(timers.isEmpty()) {
+      timers = Sets.newHashSet(0);
+    }
+    if(queries == 0) {
+      queries = 50000000l;
+    }
+    if(Strings.isNullOrEmpty(name)) {
+      name = "Test";
+    }
+
+    Set<List<Integer>> parameters = Sets.cartesianProduct(threads, depth, forkAtDepth, counters, meters,
+        histograms, timers);
+
+    Comparator<List<Integer>> comparator = new Comparator<List<Integer>>() {
```

Comment:
Also whats the reason for having this `Comparator` and the `TreeMap` as far as I can tell it only affects the order which the results are printed out.